lib/tesla/middleware/meta_logger.ex

```diff
@@ -1,134 +1,136 @@
-defmodule Tesla.Middleware.MetaLogger do
-  @moduledoc """
-  Tesla middleware to log requests and responses.
+if Code.ensure_loaded?(Tesla) do
```

Comment:
😱 this is super ugly. is there no other way? 
if not, then we should extract this module into another library and add it explicitly where we want to use it.

Reply:
Maybe a wrapper function in `Tesla.run/2`.

Reply:
I know but this is exactly what we need. We have Tesla as optional dependency and this module depends on it so when the dependency is not there we shouldn't compile it.

Reply:
I changed my mind after the discussion in Slack: https://findhotel.slack.com/archives/GJDF4TX38/p1619107278055000. 

It resolves at compile time, much better than runtime.